extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -2159,6 +2159,63 @@ describe('amp-a4a', () => {
     });
   });
 
+  describe('refresh', () => {
+    let sandbox;
+
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it('should effectively reset the slot and invoke given callback', () => {
```

Comment:
These tests are rather painful to write given how much they rely on the lifecycle methods. Before I continue, could you verify that the rest of the PR looks good, so that I can be sure the tests I'm writing make sense?